docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -164,3 +164,329 @@ Response:
   "result": 0.8
 }
 --------------------------------------------------
+
+==== Runtime field contexts
+
+Runtime field contexts execute a script used for a runtime field and return results in doc-values order.
+
+[[painless-execute-api-contexts-runtime-fields]]
+.Contexts
+[options="header"]
+|======
+| Name            | Emit Type      | Return Type
+| boolean_field   | boolean        | sorted list of booleans
+| date_field      | long (ms)      | sorted list of longs (ms)
+| double_field    | double         | sorted list of doubles
+| geo_point_field | long (encoded) | sorted list of longs (encoded)
+| ip_field        | String         | sorted list of BytesRef[] (encoded)
+| keyword_field   | String         | sorted list of Strings
+| long_field      | long           | sorted list of longs
+|======
+
+[[painless-execute-api-parameters-runtime-fields]]
+.Parameters
+[options="header"]
+|======
+| Name              | Required  | Default | Description
+| `script`          | yes       | -       | The script to execute.
+| `context`         | yes       | -       | The runtime field context the script should be executed in.
+| `context_setup`   | yes       | -       | Additional parameters to the context.
+|======
+
+Specification of the following parameters is required in `context_setup` for a runtime field context:
```

Comment:
I'd find this easier to understand if you wrote it in the active voice. "`context_setup` must contain `document` and `index`. Or something. I dunno.

Reply:
++ If you want to keep the list, you can do something like:

```suggestion
`context_setup` requires the following parameters:
```